### PR TITLE
match tuning round 3

### PR DESCRIPTION
**Pull request by sokrahta**
What's new
--------
+ support Unicode character code points
+ treat as token string: currency symbols, ampersand
+ treat as delimiter: forward slash, dot

The delimiter and token Regex classes
--------
delimiters are one or more of these:

code point | description
-------- | --------
\p{C} | control characters
\p{P} | punctuation
\p{Sm} | symbols: math
\p{Sk} | symbols: modifier
\p{So} | symbols: other
\p{Z} | separators
&&[^\&] | exclude ampersand

What's left will be included in tokens:

code point | description
-------- | --------
\p{L} | letters from any language
\p{M} | marks to be combined with another character
\p{Sc} | symbols: currency
\p{N} | numbers
by exclusion | ampersand

### CODE REVIEW

File: spandex-common/src/main/resources/reference.conf [hunk 1]
```diff
@@ -42,7 +42,7 @@ com.socrata.spandex {
              "tokenizer":{
                "word":{
                  "type":"pattern",
-                 "pattern":"\\W+",
+                 "pattern":"[\\p{C}\\p{P}\\p{Sm}\\p{Sk}\\p{So}\\p{Z}&&[^\\&]]+",
```
Comment (velvia) [inline spandex-common/src/main/resources/reference.conf]: Would you mind adding a comment to explain the pattern and why `\W+` is not adequate?  Thx.

File: spandex-common/src/main/scala/com/socrata/spandex/common/CompletionAnalyzer.scala [hunk 2]
```diff
@@ -45,7 +45,8 @@ object CompletionAnalyzer {
 
 class CompletionAnalyzer(val config: AnalysisConfig) {
   val version = Try { Version.parseLeniently(config.luceneVersion) }.getOrElse(Version.LATEST)
-  val analyzer = new PatternAnalyzer(version, Pattern.compile( """\W+"""))
+  val analyzer = new PatternAnalyzer(version, Pattern.compile(
+    """[\p{C}\p{P}\p{Sm}\p{Sk}\p{So}\p{Z}&&[^\&]]+"""))
```
Comment (velvia) [inline spandex-common/src/main/scala/com/socrata/spandex/common/CompletionAnalyzer.scala]: Hm.  This pattern is the same as the one from reference.conf ... could we just get it from there, or is this different?

File: spandex-common/src/test/scala/com/socrata/spandex/common/CompletionAnalyzerSpec.scala [hunk 3]
```diff
@@ -147,4 +155,54 @@ class CompletionAnalyzerSpec extends FunSuiteLike with Matchers with AnalyzerTes
     // the config value 32 limits to -^ plus spaces, and a little more to preserve terms
     CompletionAnalyzer.analyze(value) should equal(expectedInputValues)
   }
+
+  test("match: non-english unicode") {
+    val expectedValue = "愛"
```
Comment (velvia) [inline spandex-common/src/test/scala/com/socrata/spandex/common/CompletionAnalyzerSpec.scala]: I like this character.  :)

**Comment by velvia** [comment]
Reviewed up to 5469afe.  Looks pretty awesome overall.

File: spandex-common/src/main/scala/com/socrata/spandex/common/CompletionAnalyzer.scala (hunk 2 above)
Comment (sokrahta) [inline spandex-common/src/main/scala/com/socrata/spandex/common/CompletionAnalyzer.scala]: They happen to be the same right now, but are not coupled. And elasticsearch configuration is planned to move out of spandex configuration into a bootstrap somewhere.

File: spandex-common/src/test/scala/com/socrata/spandex/common/CompletionAnalyzerSpec.scala (hunk 3 above)
Comment (sokrahta) [inline spandex-common/src/test/scala/com/socrata/spandex/common/CompletionAnalyzerSpec.scala]: :love:

Comment (sokrahta) [inline spandex-common/src/test/scala/com/socrata/spandex/common/CompletionAnalyzerSpec.scala]: :heart:

File: spandex-common/src/main/resources/reference.conf (hunk 1 above)
Comment (sokrahta) [inline spandex-common/src/main/resources/reference.conf]: yeah no prob!

**Comment by velvia** [comment]
@sokrahta thanks, very clear comment!  
:+1:

**Pull request merged**